app/models/cart_validations.rb

```diff
@@ -105,49 +105,55 @@ def check_cookie_limit
     errors
   end
 
-  def check_max_items(count_hash, relevant, count_method)
-    # generic method for making sure the count_hash doesn't
-    # contain more than the item max when taking into account
-    # the relevant reservations
+  def check_max_ems
+    # check to make sure that the cart's EMs + the current resever's
+    # EMs doesn't exceed any limits
+    # 1 query
+
+    # get reserver's active reservations
+    source = Reservation.active.for_reserver(reserver_id)
 
     errors = []
-    count_hash.each do |item, q|
-      max = item.maximum_per_user
-
-      start_date.upto(due_date) do |d|
-        unless Reservation.send(count_method, d, item.id, relevant) + q > max
-          next
-        end
-        errors << "Only #{max} #{item.name.pluralize(max)} "\
+    get_items.each do |model, number_in_cart|
+      # get highest number of items reserved in date range of cart
+      valid = Reservation.number_for_date_range(source, start_date..due_date,
+                                                equipment_model_id: model.id,
+                                                overdue: false).max
+      valid ||= 0
+      # count overdue reservations for this eq model
+      overdue = source.count { |r| r.equipment_model == model && r.overdue }
+
+      if valid + overdue + number_in_cart > model.maximum_per_user
+        errors << "Only #{model.maximum_per_user} #{model.name.pluralize} "\
```

Comment:
Don't we need to pass `model.maximum_per_user` into `pluralize` for it to make sense? This seems like it would be a regression of #1242. We should also address this in `check_max_cat`.